### PR TITLE
Feature fds 124 documentation

### PR DESCRIPTION
**Pull request by andrewelamb**
Fixes [FDS-124](https://sagebionetworks.jira.com/jira/software/c/projects/FDS/issues/FDS-124)

[FDS-124]: https://sagebionetworks.jira.com/browse/FDS-124?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by andrewelamb** [comment]
@mialy-defelice @milen-sage @GiaJordan @linglp 
This is a first draft for the readme. This is just intended for devs, and not users. The user section will be added after this. You all are the intended audience, so if you could look this over and tell me where things aren't clear, or if there is anything I should add please let me know.

File: README.md
Comment (mialy-defelice) [inline README.md:1]: Can you format `tests/data/secrets.yml` and `.gitignore` on line 41 as inline code?

In the secrets section can you specify that the user needs to create a new project and use that new projectId?

**Review by mialy-defelice — CHANGES_REQUESTED** [review]

File: README.md
Comment (mialy-defelice) [reply README.md:1]: I also had an issue previously with `synapse client` installation, does the movement of the `all-extras` install address this?

Comment (andrewelamb) [reply README.md:1]: #183 Should address the installation issue you had earlier. Now you should be able to do 

`poetry install --all-extras`

or 

`poetry install -E synapse`

**Comment by linglp** [comment]
@andrewelamb Do you think if it will be better to use pre-commit and run linters as pre-commit hook? So that before each commit, the formatting of the code would already be taken care of.

**Comment by andrewelamb** [comment]
@linglp  Essentially if you don't have MySQL installed locally you can't install the mysql package. (I think the same is true for Postgres) This was the motivation to make it an extra. I should add a section describing this.

**Comment by andrewelamb** [comment]
@linglp I've been thinking about adding pre-commit hooks since Bruno talked about them at one point. I would have to look into them, and see if they would make sense.

**Pull request merged**